tests/storage/snapshots/conftest.py

```
@@ -149,3 +173,214 @@ def vm_restore_with_predictable_names(
+            snapshot_name=snapshot.name,
+            client=admin_client,
+        )
+        for vm, snapshot in snapshots_of_four_vms.items()
+    }
+    deadline = time.monotonic() + TIMEOUT_5MIN
+
+    def _restore_vm(restore: VirtualMachineRestore) -> VirtualMachineRestore:
+        restore.deploy()
+        remaining = deadline - time.monotonic()
+        if remaining <= 0:
+            raise TimeoutError(f"Restore {restore.name} exceeded the 5-minute budget before wait")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)